server/item/armour_trim.go

```diff
@@ -0,0 +1,48 @@
+package item
+
+import "github.com/df-mc/dragonfly/server/world"
+
+type ArmourTrim struct {
+	Template ArmourTrimTemplate
+	Material TrimMaterial
+}
+
+type TrimMaterial interface {
+	// TrimMaterial returns the material name used for reading and writing trim data.
+	TrimMaterial() string
+	// MaterialColor returns the color code used for internal text formatting. Use text.Colourf for proper formatting.
```

Comment:
You recommend to use text.Colourf although you only return the formatting character rather than the section symbol also, I wonder if we should expand gophertunnel's text package to create exported constants for each colour

Reply:
This is probably a good idea I could make a pr for this on gophertunnel